test/e2e/scheduling/priorities.go

```diff
@@ -0,0 +1,713 @@
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	priorityutil "k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/util"
+	"k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+type Resource struct {
+	MilliCPU int64
+	Memory   int64
+}
+
+var balancePodLabel map[string]string = map[string]string{"name": "priority-balanced-memory"}
+
+var podRequestedResource *v1.ResourceRequirements = &v1.ResourceRequirements{
+	Limits: v1.ResourceList{
+		"memory": resource.MustParse("100Mi"),
+		"cpu":    resource.MustParse("100m"),
+	},
+	Requests: v1.ResourceList{
+		"memory": resource.MustParse("100Mi"),
+		"cpu":    resource.MustParse("100m"),
+	},
+}
+
+// This test suite is used to verifies scheduler priority functions based on the default provider
+var _ = framework.KubeDescribe("SchedulerPriorities [Serial]", func() {
+	var cs clientset.Interface
+	var nodeList *v1.NodeList
+	var systemPodsNo int
+	var ns string
+	var masterNodes sets.String
+	f := framework.NewDefaultFramework("sched-priority")
+	ignoreLabels := framework.ImagePullerLabels
+
+	AfterEach(func() {
+	})
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace.Name
+		nodeList = &v1.NodeList{}
+
+		framework.WaitForAllNodesHealthy(cs, time.Minute)
+		masterNodes, nodeList = framework.GetMasterAndWorkerNodesOrDie(cs)
+
+		err := framework.CheckTestingNSDeletedExcept(cs, ns)
+		framework.ExpectNoError(err)
+
+		err = framework.WaitForPodsRunningReady(cs, metav1.NamespaceSystem, int32(systemPodsNo), 0, framework.PodReadyBeforeTimeout, ignoreLabels, true)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Pods should be scheduled to low resource use rate node", func() {
+		//Make sure all the schedulable nodes are balanced (have the same cpu/mem usage ratio)
+		By("Create pods on each node except the last one to raise cpu and memory usage to the same high level")
+		var expectedNodeName string
+		expectedNodeName = nodeList.Items[len(nodeList.Items)-1].Name
+		nodes := nodeList.Items[:len(nodeList.Items)-1]
+		// make the nodes except last have cpu,mem usage to 90%
+		createBalancedPodForNodes(f, cs, ns, nodes, podRequestedResource, 0.9)
+		By("Create a pod,pod should schedule to the least requested nodes")
+		createPausePod(f, pausePodConfig{
+			Name:      "priority-least-requested-mem",
+			Labels:    map[string]string{"name": "priority-least-requested-mem"},
+			Resources: podRequestedResource,
+		})
+		By("Wait for all the pods are running")
+		err := f.WaitForPodRunning("priority-least-requested-mem")
+		framework.ExpectNoError(err)
+		By("Verify the pod is scheduled to the expected node")
+		testPod, err := cs.Core().Pods(ns).Get("priority-least-requested-mem", metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		Expect(testPod.Spec.NodeName).Should(Equal(expectedNodeName))
+	})
+
+	It("Pods created by ReplicationController should spread to different node", func() {
```

Comment:
prefer to just keep one case which is enough to cover the "selector" priority's logic; for how to get selector, I'd like to dependent on unit test :).